File: llm/src/util/jwt.util.js
// src/jwtutil/jwt.util.js
const jwt = require("jsonwebtoken");

function validateToken(token) {
    const secret = process.env.JWT_SECRET;

    if (!secret) {
        throw new Error("missing JWT_SECRET");
    }

    try {
        const decoded = jwt.verify(token, secret, {
            algorithms: ["HS256"],
        });

        return decoded; // this is your Claim
    } catch (err) {
        throw new Error(err.message || "invalid token");
    }
}

module.exports = {
    validateToken,
};

File: question/src/jwtutil/jwtutil.go
package jwtutil

import (
	"fmt"
	"os"
	"questionbank/src/dto"

	"github.com/golang-jwt/jwt/v5"
)


func ValidateToken(tokenString string) (*dto.Claim, error) {
	secret := os.Getenv("JWT_SECRET")
	if secret == "" {
		return nil, fmt.Errorf("missing JWT_SECRET")
	}

	var claim dto.Claim

	// Parse the token into our custom claim struct
	token, err := jwt.ParseWithClaims(
		tokenString,
		&claim,
		func(t *jwt.Token) (interface{}, error) {
			// Check if signing method is HS256
			if _, ok := t.Method.(*jwt.SigningMethodHMAC); !ok {
				return nil, fmt.Errorf("invalid signing method")
			}
			return []byte(secret), nil
		},
	)

	if err != nil {
		return nil, err // parsing or validation failed
	}

	if !token.Valid {
		return nil, fmt.Errorf("invalid token")
	}

	return &claim, nil
}
